### ISSUE TITLE
`dotnet ef migration add` generates `DropTable()` calls for never used `ToTable()` calls

### ISSUE THREAD

**Issue by lauxjpn** [dotnet/efcore#23167]
Pomelo has some legacy migration code checks in place. We test to generate new migrations on top of legacy migrations generated for version `1.1.2` and `2.0.1` (Pomelo or EF Core, haven't checked).

These tests have been working fine since forever (including EF Core 3.1.9).
But for 5.0-rc.1, the following part of our model will generate unexpected `DROP TABLE` statements:

```c#
modelBuilder.Entity("Pomelo.EntityFrameworkCore.MySql.FunctionalTests.Models.Person", b =>
    {
        b.Property<int>("Id")
            .ValueGeneratedOnAdd();

        b.Property<string>("Discriminator")
            .IsRequired()
            .HasMaxLength(63);

        b.Property<int?>("FamilyId");

        b.Property<string>("Name");

        b.Property<int?>("TeacherId");

        b.HasKey("Id");

        b.HasIndex("FamilyId");

        b.ToTable("People");

        b.HasDiscriminator<string>("Discriminator").HasValue("Person");
    });

modelBuilder.Entity("Pomelo.EntityFrameworkCore.MySql.FunctionalTests.Models.PersonKid", b =>
    {
        b.HasBaseType("Pomelo.EntityFrameworkCore.MySql.FunctionalTests.Models.Person");

        b.Property<int>("Grade");

        b.HasIndex("Discriminator");

        b.HasIndex("TeacherId");

        b.ToTable("PersonKid");

        b.HasDiscriminator().HasValue("PersonKid");
    });
```

The model definition is represented by the following model classes:

```c#
public abstract class Person
{
	public int Id { get; set; }

	public string Name { get; set; }

	public int? TeacherId { get; set; }

	public PersonFamily Family { get; set; }
}

public class PersonKid : Person
{
	public int Grade { get; set; }

	public PersonTeacher Teacher { get; set; }
}
```

And the table looks like this:

```sql
CREATE TABLE `People` (
  `Id` int NOT NULL AUTO_INCREMENT,
  `Discriminator` varchar(63) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci NOT NULL,
  `FamilyId` int DEFAULT NULL,
  `Name` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci,
  `TeacherId` int DEFAULT NULL,
  `Grade` int DEFAULT NULL,
  `Occupation` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci,
  `OnPta` tinyint(1) DEFAULT NULL,
  PRIMARY KEY (`Id`),
  KEY `IX_People_FamilyId` (`FamilyId`),
  KEY `IX_People_Discriminator` (`Discriminator`),
  KEY `IX_People_TeacherId` (`TeacherId`),
  CONSTRAINT `FK_People_People_TeacherId` FOREIGN KEY (`TeacherId`) REFERENCES `People` (`Id`) ON DELETE RESTRICT,
  CONSTRAINT `FK_People_PeopleFamilies_FamilyId` FOREIGN KEY (`FamilyId`) REFERENCES `PeopleFamilies` (`Id`) ON DELETE RESTRICT
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```

When we generate a new migration for the current (now 5.0-rc.1 using) code, we expect no `DropTable()` calls being generated, but we get the following:

```c#
using System;
using Microsoft.EntityFrameworkCore.Metadata;
using Microsoft.EntityFrameworkCore.Migrations;

namespace Pomelo.EntityFrameworkCore.MySql.IntegrationTests.Migrations
{
    public partial class current : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropTable(
                name: "PersonKid");

            migrationBuilder.DropTable(
                name: "PersonParent");

            migrationBuilder.DropTable(
                name: "PersonTeacher");

            // Other stuff.
```

Now, I am aware that this happens because of the unnecessary `ToTable()` calls in the model definition and frankly, they shouldn't be in the legacy script in the first place. But since they are and did not generate `DropTable()` calls before, this seems to be a breaking change.

If this is expected, than you can just ignore this issue. Otherwise, unnecessary `ToTable()` calls should not generate `DropTable()` for non-existing tables.

In any case, we will remove the unnecessary `ToTable()` calls from our legacy scripts.

### Include provider and version information

EF Core version: 5.0-rc.1
Database provider: Pomelo.EntityFrameworkCore.MySql
Target framework: .NET 5.0
Operating system: Windows 10

**Comment by bricelam** [comment]
We fixed the bug of scaffolding the invalid ToTable call in EF Core 2.2, so this only affects users who upgrade to EF Core 5.0 and haven't added migrations since EF Core 2.1.

We should definitely fix this for #18557, but let's discuss whether we should patch 5.0.

**Comment by ajcvickers** [comment]
Note for triage: too late to patch for 5.0.
